xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-requiredrights/xwiki-platform-security-requiredrights-default/src/main/java/org/xwiki/platform/security/requiredrights/internal/RequiredRightsChangedFilter.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.platform.security.requiredrights.internal;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.document.DocumentAuthors;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.platform.security.requiredrights.RequiredRightAnalysisResult;
+import org.xwiki.security.authorization.AuthorizationManager;
+import org.xwiki.security.authorization.Right;
+import org.xwiki.user.UserReferenceSerializer;
+
+import com.xpn.xwiki.XWikiContext;
+
+import static org.xwiki.model.EntityType.OBJECT;
+import static org.xwiki.model.EntityType.OBJECT_PROPERTY;
+
+/**
+ * Takes a list of {@link RequiredRightAnalysisResult} and filter them according to a set of {@link DocumentAuthors}.
+ *
+ * @version $Id$
+ * @since 15.9RC1
+ */
+@Component(roles = RequiredRightsChangedFilter.class)
+@Singleton
+public class RequiredRightsChangedFilter
+{
+    @Inject
+    private AuthorizationManager authorizationManager;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    @Named("document")
+    private UserReferenceSerializer<DocumentReference> userReferenceSerializer;
+
+    /**
+     * Takes a list of {@link RequiredRightAnalysisResult} and filter them according to a set of
+     * {@link DocumentAuthors}.
+     *
+     * @param authors the document authors to use for the filtering
+     * @param resultList the list of required rights analysis result to filter
+     * @return the filtered list of required rights results, according to the provided document authors
+     */
+    public RequiredRightsChangedResult filter(DocumentAuthors authors,
```

Comment:
So that one is not easy to read, it's very compact might worth a bit of refactoring / comment. Also I'm not sure to understand why it's not part of the API? Sounds like a common need no ?

Reply:
Regarding the API, I'm not sure. For now this is only used by internal component and I'm not sure we want to expose this part to extensions.

Reply:
Ok so better keep it internal indeed for now.